bfcfg

```diff
@@ -1287,14 +1293,578 @@ sb_cfg()
+# For example, val=128, len=4 -> \x80\x00\x00\x00
+to_bytes()
+{
+  local val=$1
+  local len=$2
+  local bytes=$(printf "%0$(( $len * 2 ))X" "$val" | xxd -p -c 2 | sed '1!G;h;$!d' | tr -d '\n' | xxd -r -p | sed -e 's/[0-9A-F]\{2\}/&\\x/g' -e 's/\\x$//')
+
+  printf "\\x$bytes"
+}
+
+# Converts bytes read from input file to integer.
+# Bytes are swapped before conversion.
+# For example, 0A -> 10
+to_integer()
+{
+  local in_file=$1
+  local offset=$2
+  local size=$3
+  local value=0
+
+  value=$(dd if=${in_file} skip=${offset} count=${size} bs=1 2> /dev/null | xxd -p -c 1 | sed '1!G;h;$!d' | tr -d '\n')
+  echo $(( 16#$value ))
+}
+
+arm_cfg_get_value()
+{
+  local out_file=$1
+  local name=$2
+  local offset=$3
+  local size=$4
+  local in_file=$5
+  local value=0
+
+  if [ -f "${in_file}" ]; then
+    if [[ "$name" == "BOOT"* ]] \
+      || [ "$name" = "PXE_DHCP_CLASS_ID" ] \
+      || [ "$name" = "BF_BUNDLE_VERSION" ]; then
+      value=$(dd if=${in_file} skip=${offset} count=${size} bs=1 2> /dev/null | tr -d '\0')
+      if [[ "$name" == "BOOT"*"_DESC" ]] \
+      || [[ "$name" == "BOOT"*"_ARGS" ]] \
+      || [[ "$name" == "BOOT"*"_DEVPATH" ]]; then
+        if [ -n "$value" ]; then echo "$name='$value'" >> ${out_file}; fi
+      else
+        if [ -n "$value" ]; then echo "$name=$value" >> ${out_file}; fi
+    fi
+
+    elif [[ "$name" == "SYS_"* ]] \
+      || [ "$name" = "FACTORY_DEFAULT_DHCP_BEHAVIOR" ]; then
+      value=$(dd if=${in_file} skip=${offset} count=${size} bs=1 2> /dev/null | tr -d '\0')
+      if [ "$value" = "1" ]; then echo "$name=TRUE" >> ${out_file}; fi
+      if [ "$value" = "0" ]; then echo "$name=FALSE" >> ${out_file}; fi
+
+    elif [ "$name" = "MFG_PLAT_MODE" ]; then
+      value=$(to_integer ${in_file} ${offset} $size)
+      if [ $value -eq 1 ]; then echo "$name=DPU" >> ${out_file}; fi
+      if [ $value -eq 2 ]; then echo "$name=NIC" >> ${out_file}; fi
+
+    elif [ "$name" = "NET_RSHIM_MAC" ]; then
+      value=$(dd if=${in_file} skip=${offset} count=${size} bs=1 2> /dev/null | xxd -p)
+      mac=$(echo $value | sed -e 's/../&:/g' -e 's/:$//')
+      echo "$name=$mac" >> ${out_file}
+
+    elif [ "$name" = "NET_DHCP_IPV6_DUID" ]; then
+      value=$(to_integer ${in_file} ${offset} $size)
+      if [ $value -eq 1 ]; then echo "$name=UUID" >> ${out_file}; fi
+      if [ $value -eq 2 ]; then echo "$name=LLT" >> ${out_file}; fi
+
+    fi
+  fi
+
+  return
+}
+
+arm_sys_cfg_set_value()
+{
+  local out_file=$1
+  local name=$2
+  local offset=$3
+  local size=$4
+  local value=$5
+
+  if [ -n "${value}" ] && [ -f "${out_file}" ]; then
+    if [ "$name" = "NET_DHCP_IPV6_DUID" ]; then
+      if [ "${value^^}" == "UUID" ]; then printf "\x01" | dd of="${out_file}" seek="${offset}" bs=1 count=${size} conv=notrunc 2> /dev/null; fi
+      if [ "${value^^}" == "LLT" ]; then printf "\x02" | dd of="${out_file}" seek="${offset}" bs=1 count=${size} conv=notrunc 2> /dev/null; fi
+    elif [ "$name" = "MFG_PLAT_MODE" ]; then
```

Comment:
I think we need to handle SYS_ENABLE_SPCR explicitly as well, since the acceptable values are 0,1,2 and not just TRUE or FALSE, right ?

Reply:
Ack, Sounds good.